unit_tests.py

```diff
@@ -0,0 +1,167 @@
+        """
+            Check if the errors reported by grpcat are as expected.
+        """
+
+        #too many or few args
+        test_args = [
+        [ 'Too Few', '/hello.Hello/GetHello',
+        'GetHelloRequest'],
+        ['Too Many', '5555', '/test2/Get-Test2',
+        'GetRequest2', 'extra']
+        ]
+
+        expected_errors = [
+        "usage: unit_tests.py [-h] host port rpc message_type\n"
+        "unit_tests.py: error: the following arguments are required:"
+        " message_type\n",
+        "usage: unit_tests.py [-h] host port rpc message_type\n"
+        "unit_tests.py: error: unrecognized arguments: extra\n",
+        ]
+
+        #save original out streams
+        old_stdout = sys.stderr
+        old_stderr = sys.stdout
+        #throw away help output
+        sys.stdout = out = open(os.devnull, "w")
+
+        for i in range(len(test_args)):
+            #redirect error output for comparison
+            sys.stderr = err = StringIO()
+
+            sys.argv[1:] = test_args[i]
+            #catch the parse arg exception
+            try:
+                args = grpcat.parse_args()
+                gprcat.run(args)
+            except:
+                pass
+
+            self.assertEqual(err.getvalue(),expected_errors[i])
+
+        out.close()
+        #resume normal output
+        sys.stdout = old_stdout
+        sys.stderr = old_stderr
+
+
+        def test_arg_format_errors(self):
+            """
+                Test if the correct formatting exception is raised when
+                the target service is not in the form
+                /<proto definition name>.<service>/<method>
+            """
+
+            test_args = [
+                [ 'RPC Format Error', '/hello.HelloGetHello',
+                'GetHelloRequest'],
+                ['localhost', '4444', '/helloHello/GetHello',
+                'GetHelloRequest'],
+                ['localhost', '4444', '/hello-Hello/GetHello',
+                'GetHelloRequest'],
+                ['localhost', '4444', '\hello.Hello\GetHello',
+                'GetHelloRequest']
+            ]
+
+            for t in test_args:
+                sys.argv[1:] = test_args[i]
+
+                args = grpcat.parse_args()
+                self.assertRaises(ValueError, grpcat.run(args))
+
+
+class TestRPCResponse(unittest.TestCase):
+    """
+        Tests to check server response.
+    """
+
+
+    def test_failed_connection(self):
+        """
+            Test whether the connection to the RPC server fails.
+        """
+
+        sys.argv[1:] = ['localhost', '4444', '/hello.Hello/GetHello',
+        'GetHelloRequest']
+
+        fail_message = ("<_Rendezvous of RPC that terminated with (StatusCode."
+        "UNAVAILABLE, Connect Failed)>")
+
+        args = grpcat.parse_args()
+        try:
+            grpcat.run(args)
+        except Exception as e:
+            self.assertEqual(fail_message, str(e))
+
+
+
+    def test_successful_connection(self):
```

Comment:
The test checks whether grpcat raises any exceptions, if there are no exceptions raised by grpcat, there should be a response from the server.